generator/lib/src/writer/writer.dart

```diff
@@ -328,7 +328,9 @@ class Writer {
       _w.write('final Upsert upsert = upserter');
       _w.writeln(
           '.setMany(toSetColumns(model, only: only, onlyNonNull: onlyNonNull));');
-      _w.writeln('return adapter.upsert(upsert, withConn: withConn);');
+      _w.writeln('var id = await adapter.upsert(upsert, withConn: withConn);');
```

Comment:
be careful here because it's not really the id that is returned here, if you have an auto incremented primary key id it will match the that value BUT if you don't have it auto incremented it will return the rowid and not your id, it's fine to let this variable as "id" because for other database it's how it works but not for sqlite, we might want to add a comment on the generated code to explain that for the one who will read it ^^